lib/pure/asyncdispatch.nim

```diff
@@ -262,15 +280,110 @@ when defined(windows) or defined(nimdoc):
     PostCallbackDataPtr = ptr PostCallbackData
 
     AsyncEventImpl = object
-      hEvent: Handle
-      hWaiter: Handle
-      pcd: PostCallbackDataPtr
+      triggered: bool
+      when compileOption("threads"):
+        eventLock: RLock
+      p: PDispatcher
+      vFD: VirtualFD
+      cb: Callback
```

Comment:
`Callback` is `closure` so it still from thread-local heap, and `PDispatcher` is also from thread-local heap.

Reply:
The only field that is accessed from multiple threads is the `triggered` bool, and it is protected by `eventLock`.
The other fields are never accessed from a different thread.